Clarifications in contributors guide (#7595)
pydata/xarray: doc/contributing.rst

This creates the directory `xarray` and connects your repository to
the upstream (main project) *xarray* repository.

> Creating a branch

Reviewer: In a later PR, IMO we should link out to the Github docs more for the  general stuff. They put a bunch of effort in to making it easy. E.g. https://docs.github.com/en/get-started/quickstart/github-flow

xarray uses several tools to ensure a consistent code format throughout the project:

- `Black <https://black.readthedocs.io/en/stable/>`_ for standardized
  code formatting,
- `blackdoc <https://blackdoc.readthedocs.io/en/stable/>`_ for
  standardized code formatting in documentation,

> - `flake8 <http://flake8.pycqa.org/en/latest/>`_ for general code quality (we now use `ruff <https://github.com/charliermarsh/ruff/>`_ to apply these checks),
> - `isort <https://github.com/timothycrosley/isort>`_ for standardized order in imports
>   (again we use `ruff <https://github.com/charliermarsh/ruff/>`_ to apply this formatting),

Reviewer: ```suggestion
- `ruff <https://github.com/charliermarsh/ruff/>`_ for code quality checks and standardized order in imports
```

> - `flake8 <http://flake8.pycqa.org/en/latest/>`_ for general code quality (we now use `ruff <https://github.com/charliermarsh/ruff/>`_ to apply these checks),
> - `isort <https://github.com/timothycrosley/isort>`_ for standardized order in imports
>   (again we use `ruff <https://github.com/charliermarsh/ruff/>`_ to apply this formatting),
> - `absolufy-imports <https://github.com/MarcoGorelli/absolufy-imports>`_ for absolute instead of relative imports from different files,

Reviewer: ```suggestion
- `ruff <https://github.com/charliermarsh/ruff/>`_ for code quality checks and standardized order in imports
- `absolufy-imports <https://github.com/MarcoGorelli/absolufy-imports>`_ for absolute instead of relative imports from different files,
```
